FWCore/Common/interface/CudaDecls.h

```diff
@@ -0,0 +1,12 @@
+#ifndef FWCore_Common_interface_CudaDecls_h
+#define FWCore_Common_interface_CudaDecls_h
+
+#ifdef __CUDACC__
+    #define HOST __host__
+    #define DEVICE __device__
+#else
+    #define HOST
+    #define DEVICE
+#endif
```

Comment:
Maybe `HeterogeneousCore/CUDACore` or `HeterogeneousCore/CUDAUtilities` would be a better place? (unless we really have to place these in deeply core classes; probably if really needed in `DataFormats` the best place would be `DataFormats/Common`.